### PR TITLE
create new jenkins pipeline

### LINKED ISSUE #695: CI : Create new Pipelines for Testing and Release
Currently we have only one pipeline job in Jenkins to support build and running smoke tests, 
We need to a few more pipelines added to support Rook release. We need pipeline jobs for the following

1. Testing -End to End and functional testing
1. Long haul testing
1. Load and performance testing. 
1. Release/publish.

### PR DESCRIPTION
**Pull request by dangula**
- complete regression
 - longhaul testing
[skip ci]
fix #695

### CODE REVIEW

**Comment by dangula** [comment]
I started out with one script with regression and long haul, but split them mainly because of 
a.) I an planning be using different node labels for regression and long haul - so regression can run on smaller VMS and long haul can be run on bigger VM with lots of disks.   And 
b.) long haul has more parameters when compared to regression, and I didn't want to confuse the user with lots of options and when to use what. Jenkins shows all parameters available to the build by default. 
and c.) I thought it would to better to have 2 pipelines so we can look at the history of regression and load test separately. eg, we might have a minor release where we only run full regression
But I totally get combining them for maintenance reasons - I was split 50/50 between both and decided to separate them(just like tag and release pipelines which are very similar but different purpose.

**Comment by dangula** [comment]
@aramisjohnson I split the pipelines into 3 pipelines now, One to set up k8s and rook images under test, 2. run the full regression and 3 run a long-haul test.  There is very little duplication between the pipeline now.

File: tests/pipeline/Jenkinsfile.k8sSetup [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                                sh '''#!/bin/bash
+                                      echo " setting up node 2"
+                                      export KUBE_VERSION='''+ "${env.kube_version}" + '''
+                                      tests/scripts/kubeadm.sh install node ''' + "${token}"
+                            }
+                            sh '''#!/bin/bash
+                                  export KUBECONFIG=$HOME/admin.conf
+                                  tests/scripts/kubeadm.sh wait 3'''
+                        }
+                    }finally{
```
Comment (aramisjohnson) [inline tests/pipeline/Jenkinsfile.k8sSetup]: You have this finally clause that is really not doing anything, remove?

Comment (dangula) [reply tests/pipeline/Jenkinsfile.k8sSetup]: missed cleaning it up after refactor. Should be good now.

**Pull request merged**